Makefile

```diff
@@ -1,12 +1,30 @@
-all: build
-
 FLAGS =
-COMMONENVVAR = GOOS=$(shell uname -s | tr A-Z a-z) GOARCH=$(subst x86_64,amd64,$(patsubst i%86,386,$(shell uname -m)))
 BUILDENVVAR = CGO_ENABLED=0
 TESTENVVAR = 
 REGISTRY = gcr.io/google_containers
 TAG = $(shell git describe --abbrev=0)
 PKGS = $(shell go list ./... | grep -v /vendor/)
+ARCH ?= $(shell go env GOARCH)
+ALL_ARCH = amd64 arm arm64 ppc64le s390x
+
+
+IMAGE = $(REGISTRY)/kube-state-metrics
+MULTI_ARCH_IMG = $(IMAGE)-$(ARCH)
```

Comment:
Newer versions of the docker image/registry format actually allow having the same repository for multiple arches iirc, would it be possible to use that here instead, as this is otherwise a breaking change compared to the previously distributed image repository, also is the busybox base actually necessary? Seems like a significant change compared to previously used `scratch`.

Reply:
@brancz I guess you are talking about manifest list, I don't think that has been implemented for gcr repo yet! So, until then we need to have different images for different arch.
I have removed busybox and put the scratch back.

Reply:
After some Googling, it turns out [the tools to create and push these manifest lists don't exist yet, but support for pull a manifest list has been aded to Docker 1.10](https://github.com/moby/moby/pull/18785). Also, there is a [tool](https://github.com/estesp/manifest-tool) to create and push manifest. But it is not the Docker official one. The tools for supporting manifest has not been supported well for now and @Pensu has added a compatibility to push `kubernetes/kube-state-metrics:v1.0.1` when arch is 'amd64'.

So, this should be fine. @brancz WDYT.